src/what-the-graphlet.c

```diff
@@ -0,0 +1,169 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "what-the-graphlet.h"
+
+
+
+unsigned long long _canonical_count[] =
+	{0, 1, 2, 4, 11, 34, 156, 1044, 12346, 274668, 12005168, 1018997864};
+
+Gint_type map_non_canon[MAXK+1][MAX_SIZE];
+
+Gint_type map_canon[MAXK+1][MAX_SIZE];
+unsigned char map_permutation[MAXK+1][MAX_SIZE][MAXK+1];
+
+Gint_type ordinal_to_canon[MAXK+1][MAX_CANON];
+
+
+
+
+
+void read_maps(int max_k) {
+    char filename[100];
+    Gint_type cnt;
+    Gordinal_type cannon_cnt;
+    char buf[BUFSIZ];
+
+    for(int k = 3; k <= max_k; k++) {
+        sprintf(filename, "%s%s%d.txt", CANON_MAP_FOLDER, "smaller_canon_map", k);
+        FILE* file = fopen(filename, "r");
+
+        cnt = 0, cannon_cnt = 0;
+        while(fgets(buf, sizeof(buf), file)) {
+            sscanf(buf, "%llu\t%llu\t%s\t%*s", &map_non_canon[k][cnt], &map_canon[k][cnt], map_permutation[k][cnt]); // Need to change formates
```

Comment:
Store the return value of sscanf, and ensure it returns the correct number--just like the errors you were fighting with yesterday in compute-alphas.

Reply:
I don't know how to do it